src/mscorlib/src/System/String.cs

```diff
@@ -1695,29 +1695,104 @@ private String CtorCharCount(char c, int count)
         private static unsafe int wcslen(char *ptr)
         {
             char *end = ptr;
+            
+            // TODO: Test to see if switching to something like
+            // the 64-bit version would be beneficial on x86 as well
+
+            // First make sure our pointer is aligned on a word boundary
+            int alignment = IntPtr.Size - 1;
+
+            // If ptr is at an odd address, this loop will simply iterate all the way
```

Comment:
This comment doesn't look right.  This isn't checking to see if ptr is at an odd (as in not even) address, but rather whether it's at an aligned address.  And if it isn't aligned, it's not iterating "all the way" (assuming that means to the end), but rather only until it hits an aligned address, unless it it's very short and it gets to the end before it hits an aligned address.